ui/Stars/StarsManager.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import HalfStar from './HalfStar';
+import FullStar from './FullStar';
+import GrayStar from './GrayStar';
+
+interface StarsProps {
+  rating: number;
+}
+
+const StarsManager: React.FC<StarsProps> = ({ rating }) => {
+  const stars = [...Array.from({ length: 5 })].map((_, index) => {
+    if (index < Math.floor(rating)) {
+      return <FullStar key={index} />;
+    }
+    if (index === Math.floor(rating)) {
+      if (rating % 1 === 1) {
+        return <FullStar key={index} />;
+      } else if (rating % 1 > 0.5) {
+        return <HalfStar key={index} />;
+      } else return <GrayStar key={index} />;
+    } else return <GrayStar key={index} />;
+  });
+  return <div className="flex">{stars}</div>;
```

Comment:
no te IFy tutaj to rzeczywiście brzydkie :)

Reply:
moja propozycja na te gwiazdki:
```tsx
const MAX_STARS = 5;
function StarsMenager ({ rating }) {

  return [...Array.from({ length: MAX_STARS })].map((_, index) => {
    const { type } = getStarType(rating, index)
    return <Star type={type} rating={rating} key=index />
  }
}
```
i cała logika w tej funkcji, która zwraca odpowiedni typ
/stars/utils/getStarType
```tsx
export enum StarTyp = {
Full,
Empty,
Half
}
export default getStarType(rating, index): StarTyp {
...
}
```

w Star:
```tsx
if (type === StarType.Full ) {
  return ...
}
if (type === StarType.Empty ) {
  return ...
}
if (type= StarType.Half) {
 const decimal = // oblicz na podstawie rating i index (albo inaczej)

}
```

Reply:
zanim to doczytałem zrobiłem swoją refaktoryzację. I biję się teraz z myślami. W mojej jest trochę mniej kodu, ale zostają dwa dosyć złożone ify. Zerknij potem i napisz czy przepisywać dalej...